src/pager.c

```diff
@@ -7391,7 +7403,9 @@ int sqlite3PagerOkToChangeJournalMode(Pager *pPager){
 i64 sqlite3PagerJournalSizeLimit(Pager *pPager, i64 iLimit){
   if( iLimit>=-1 ){
     pPager->journalSizeLimit = iLimit;
-    sqlite3WalLimit(pPager->pWal, iLimit);
+    if (pagerUseWal(pPager)) {
```

Comment:
Why do we have to check whether pWal is NULL after virtualising the WAL interface?

Reply:
Previously, sqlite3WalLimit simply checked if `pPager->pWal` is not `null` and that was sufficient. Now it is not, because WAL support can even be compiled out by defining `SQLITE_OMIT_WAL`, and then pWalMethods is `null` and we can't call the virtual interface, because it does not exist. An alternative would be to spray `#ifdef`s on each call site, but I find `if` + `pagerUseWal` more readable.

Reply:
ah wait, it's not enough, with SQLITE_OMIT_WAL this code won't compile, because the pWalMethods field is not even there. I'll recompile with SQLITE_OMIT_WAL and double check all the missing places, and fix accordingly